firmware/config/stm32h7ems/efifeatures.h

```diff
@@ -49,3 +49,16 @@
 #define LUA_USER_HEAP 100000
 #undef LUA_SYSTEM_HEAP
 #define LUA_SYSTEM_HEAP 100000
+
+// UART driver not implemented on H7, but the serial driver is...
+#undef TS_NO_PRIMARY
+#undef TS_PRIMARY_UART
+#define TS_PRIMARY_SERIAL SD3
+
+// todo: start using consoleSerialTxPin? Not sure
+#undef EFI_CONSOLE_TX_BRAIN_PIN
+#define EFI_CONSOLE_TX_BRAIN_PIN GPIOD_8
```

Comment:
pins should probably go in the `board.mk` for the respective nucleo board

Reply:
since right now these will get used for /every/ H7 board, these could cause pin conflicts with other boards that use PD8/PD9

Reply:
This somewhat mimics what happens on F4, which has
```
#ifndef TS_PRIMARY_UART                                                                             
#define TS_PRIMARY_UART UARTD3                                                                      
#endif                                                                                              
```
and friends.  For some reason the F7/H7 more forcefully prevent this default.  I can move it but there's a lot of goofiness about TS_PRIMARY_UART vs TS_PRIMARY_SERIAL so getting the default to work correctly when one is defined and the other might be defined later is tricky.  The whole thing is just insane, the HAL layer should export available serial ports whether UART or Serial or whatever, then we'd just need one TS_PRIMARY_FOO variable to define which one to use.... or have some sentinel value like OFF to disable.

Reply:
If you think this is insane, you should've seen what it replaced!

Reply:
And I don't mean moving the UARTD3 part - I mean moving the GPIOD_8 part.  Right now this will cause a pin conflict on an H7 proteus - those pins are used for electronic throttle control.

Reply:
changed how ports are defined; TS_NO_PRIMARY is now a 0-or-1 define, not a is-or-is-not-defined.  TS_PRIMARY_UART and TS_PRIMARY_SERIAL are folded into TS_PRIMARY_PORT so there's no confusion over ordering.

Moved defines to nucleo board.mk, though it seems most boards already define TS_NO_PRIMARY etc so perhaps the default of "on" was ok.  But putting it in board dir does make more sense, it's a board configuration.